dev/AnimatedVisualPlayer/AnimatedVisualPlayer.cpp

```diff
@@ -821,12 +827,13 @@ void AnimatedVisualPlayer::CreateAnimations() {
 }
 
 void AnimatedVisualPlayer::DestroyAnimations() {
-    if (!m_isAnimationsCreated)
+    if (!m_isAnimationsCreated || m_animatedVisual == nullptr)
     {
         return;
     }
 
     // Call RequestCommit to make sure that previous compositor calls complete before destroying animations.
+    // RequestCommitAsync is available only for RS4+ but m_animatedVisual is not null guarantees that we are at least RS5+
```

Comment:
How is this the case? Don't the visuals get created at startup? and then I could call destroy animations on this and hit this code path?

Reply:
Destroy and Create are private inside AVP, we are exposing only AnimationOptimization property